narwhal/primary/src/core.rs

```diff
@@ -382,7 +384,7 @@ impl Core {
         // that are stored in the header store. This strategy can be used to re-deliver votes to
         // ensure progress / liveness.
         self.vote_digest_store
-            .write(
+            .async_write(
```

Comment:
I think this be a `sync_write` to make sure we persist the vote before going a head

Reply:
fixed!